apps/ui/components/calendar/ReservationCalendarControls.tsx

```diff
@@ -312,323 +269,79 @@ const TogglerLabelContent = ({
   );
 };
 
-const ReservationCalendarControls = <T extends Record<string, unknown>>({
+const ReservationCalendarControls = ({
   reservationUnit,
-  initialReservation,
-  setInitialReservation,
-  isSlotReservable,
-  isReserving,
-  setCalendarFocusDate,
-  activeApplicationRounds,
-  createReservation,
-  setErrorMsg,
-  handleEventChange,
   mode,
-  customAvailabilityValidation,
   shouldCalendarControlsBeVisible,
   setShouldCalendarControlsBeVisible,
-  apiBaseUrl,
   isAnimated = false,
-}: Props<T>): JSX.Element => {
-  const { t, i18n } = useTranslation();
-
-  const { begin, end } = initialReservation ?? {};
-  const {
-    minReservationDuration,
-    maxReservationDuration,
-    reservationStartInterval,
-  } = reservationUnit;
-
-  const durationOptions = useMemo(() => {
-    const options = getDurationOptions(
-      minReservationDuration ?? 0,
-      maxReservationDuration ?? 0,
-      reservationStartInterval,
-      t
-    );
-    return [{ value: "0:00", label: "" }, ...options];
-  }, [
-    minReservationDuration,
-    maxReservationDuration,
-    reservationStartInterval,
-    t,
-  ]);
-
-  const [date, setDate] = useState<Date | null>(new Date());
-  const [startTime, setStartTime] = useState<string | null>(null);
-  const [duration, setDuration] = useState<OptionType | null>(
-    durationOptions[0]
+  reservationForm,
+  durationOptions,
+  focusSlot,
+  startingTimeOptions,
+  submitReservation,
+  LoginAndSubmit,
+}: Props): JSX.Element => {
+  const { t } = useTranslation();
+  const { watch, handleSubmit } = reservationForm;
+  const { start, end } = focusSlot ?? {};
+  const formDate = watch("date");
+  const formDuration = watch("duration");
+  const date = new Date(formDate ?? "");
+  const dateValue = useMemo(() => new Date(formDate ?? ""), [formDate]);
+  const focusDate = useMemo(
```

Comment:
Date format? this is not valid if Date is in UI format (only ISO).

Reply:
Oh, that's right. Changed it to use the `fromUIDate()` helper function.